### PR TITLE
feat: add abilty to use button as A-tag button with href

### PR DESCRIPTION
**Pull request by REJack**

### CODE REVIEW

**Comment by ph7jack** [comment]
@REJack Thanks for your contribution. 
I'm making a code review

File: resources/views/components/button.blade.php [hunk 1]
```diff
@@ -1,6 +1,7 @@
-<button {{ $attributes->merge([
+@if(isset($href))
```
Comment (ph7jack) [inline resources/views/components/button.blade.php]: Don't need to use an if to switch between `a` and `button` tags, just replace the HTML tag
`<{{ $tag }} ..... > .... </{{ $tag }}>`

So, we can use
`<x-button tag="a" href="google.com" .... />`

File: src/View/Components/BaseButton.php [hunk 2]
```diff
@@ -106,6 +110,7 @@ protected function mergeData(array $data): array
         $attributes       = $data['attributes'];
         $data['disabled'] = (bool) $attributes->get('disabled');
         $data['classes']  = $this->getClasses();
+        $data['href']     = $this->href;
```
Comment (ph7jack) [inline src/View/Components/BaseButton.php]: Why did you add it? Don't need to pass the class properties to `$data`
```suggestion
```

File: resources/views/components/button.blade.php (hunk 1 above)
Comment (ph7jack) [reply resources/views/components/button.blade.php]: Add a tag option

File: tests/Browser/Button/view.blade.php [hunk 3]
```diff
@@ -45,4 +45,13 @@
...
     <x-button outline dark label="dark" right-icon="home" />
+
+    <x-button href="#" label="Label" icon="home" />
+    <x-button href="#" primary label="primary" icon="home" />
+    <x-button href="#" secondary label="secondary" icon="home" />
+    <x-button href="#" disabled positive label="positive" icon="home" />
+    <x-button href="#" disabled negative label="negative" icon="home" />
+    <x-button href="#" warning label="warning" icon="home" />
+    <x-button href="#" info label="info" icon="home" />
+    <x-button href="#" dark label="dark" icon="home" />
```
Comment (ph7jack) [inline tests/Browser/Button/view.blade.php]: Missing a test to assert if the href can is rendered

**Review by ph7jack — CHANGES_REQUESTED** [review]
Thanks for your contribution

**Comment by REJack** [comment]
I've added a correct test and changed the approach, but without the "tag"-attribute IMHO it's odd and not really needed since you need a `href` for `a`-tags and they will not used on `button`-tags.

**Comment by bernhardh** [comment]
Would be great, if this could be merged

**Comment by REJack** [comment]
> Would be great, if this could be merged

Me 2 but I think he will publish his button color changes, before merging my PR 😄 then maybe I would need to add/change stuff

**Review by ph7jack — APPROVED** [review]
Thanks for this contribution. Sorry for my long time to review

**Pull request merged**